client/src/client.ts

```diff
@@ -2438,7 +2442,17 @@ export abstract class BaseLanguageClient {
 		this._trace = value;
 		this.onReady().then(() => {
 			this.resolveConnection().then((connection) => {
-				connection.trace(value, this._tracer);
+				connection.trace(value, this._traceFormat, this._tracer);
+			})
+		}, () => {
+		});
+	}
+
+	public set traceFormat(value: TraceFormat) {
```

Comment:
Would it make sense to fold that into trace as an optional param. If not provided it is text. I doubt that users set this independently.

Reply:
Sure, since `trace` function already covers the format as an optional param. However I'm wondering if you mind adding `traceFormat` as one of the `LanguageClientOptions`? I feel this is something people want to configure on LS Client init but not setting it each time.